src/hooks/voucherHooks.ts

```diff
@@ -46,3 +49,60 @@ export const useVoucher = (): {
     memberId
   };
 };
+
+export const useRedeemLocalVouchers = () => {
+  const { isConnected, isMainserverUp } = useContext(NetworkContext);
+  const [isLoading, setIsLoading] = useState(true);
+  const [isError, setIsError] = useState(false);
+
+  const [redeemQuotaOfVoucher] = useMutation(REDEEM_QUOTA_OF_VOUCHER);
+
+  const redeemLocalVouchers = useCallback(async () => {
+    setIsLoading(true);
+    setIsError(false);
+
+    try {
+      const voucherTransactions = (await readFromStore(VOUCHER_TRANSACTIONS)) || [];
+      let error = false;
+
+      if (voucherTransactions.length) {
+        for (const { voucherId, memberId, quantity, deviceToken } of voucherTransactions) {
+          try {
+            await redeemQuotaOfVoucher({
+              variables: {
+                deviceToken,
+                quantity,
+                voucherId,
+                memberId
+              }
+            });
+          } catch (e) {
+            console.error(e);
+            error = true;
+            break;
+          }
+        }
+        if (!error) {
+          await addToStore(VOUCHER_TRANSACTIONS, []);
+        }
+      }
+    } catch (e) {
+      console.error(e);
+      setIsError(true);
+    } finally {
+      setIsLoading(false);
+    }
+  }, []);
+
+  useEffect(() => {
+    if (isConnected && isMainserverUp) {
+      redeemLocalVouchers();
+    }
+  }, [isConnected, isMainserverUp, redeemLocalVouchers]);
+
+  return {
+    refresh: redeemLocalVouchers,
```

Comment:
why is this named `refresh`?

Reply:
There's no reason. Because I copy and paste.